helix-term/src/keymap.rs

```diff
@@ -198,13 +198,35 @@ impl<'de> serde::de::Visitor<'de> for KeyTrieVisitor {
     where
         M: serde::de::MapAccess<'de>,
     {
+        let mut label = "";
+        let mut command = None;
         let mut mapping = HashMap::new();
         let mut order = Vec::new();
-        while let Some((key, value)) = map.next_entry::<KeyEvent, KeyTrie>()? {
-            mapping.insert(key, value);
-            order.push(key);
+
+        while let Some(key) = map.next_key::<&str>()? {
+            match key {
+                "label" => label = map.next_value::<&str>()?,
+                "command" => command = Some(map.next_value::<MappableCommand>()?),
+                _ => {
+                    let key_event = key.parse::<KeyEvent>().map_err(serde::de::Error::custom)?;
+                    let key_trie = map.next_value::<KeyTrie>()?;
+                    mapping.insert(key_event, key_trie);
+                    order.push(key_event);
+                }
+            }
+        }
+
+        match command {
+            None => Ok(KeyTrie::Node(KeyTrieNode::new(label, mapping, order))),
+            Some(MappableCommand::Typable { name, args, .. }) if !label.is_empty() => {
```

Comment:
If this happens and order is not-empty should an error be given? (Instead of silently ignoring sub-fields)?

Reply:
IMO this one is up to the helix maintainers. I'm not sure if there's a general policy/pattern within the project for raising errors on ambiguous input or picking a valid path to take priority.

Reply:
I think an error would be appropriate here - I think it would be confusing to silently pick a path